### PR TITLE
run by run v3 with font removed

### PR DESCRIPTION
**Pull request by ssedd1123**
It should be pure python code now.

### CODE REVIEW

**Review by marrbnl — APPROVED** [review]

**Comment by starsdong** [comment]
Any comment from your side, Grigory and Zach? Thanks

File: StRoot/PWGTools/BadRunQA/README.md [hunk 1]
```diff
@@ -1,78 +1,97 @@
...
+This software is designed to segment time series data points using a change point detection algorithm. The algorithm detects changes in the mean and variance of the data points and segments the data into different segments based on these changes. The software then removes outliers in each segment by removing points that are beyond a certain factor (default is 5) standard deviations from the mean of each segment. 
+The code will iteration on both its change point detection and outlier rejection until no more outliers are discovered.
+
+This code works on both Linux and Windows. It should work in Mac but it has not been tested.
+
+## Installation
+
+To install the software, follow these steps:
+The EASY way (Linux only. For Windows and Mac, use the recommended way):
+1. Run the following command: `python3 -m pip install --user matplotlib ruptures pandas numpy scipy pyfiglet uproot scikit-learn`
```
Comment (plexoos) [inline StRoot/PWGTools/BadRunQA/README.md]: Usually it helps to freeze the dependencies in `requirements.txt` (or in `pyproject.toml`). It should help with reproducibility of results as well.

Comment (ssedd1123) [reply StRoot/PWGTools/BadRunQA/README.md]: I have updated my repo to include requirements.txt for Linux. For some reasons the packages must be installed with anaconda on Windows so I don't have one for Windows. 

I have also updated my run-by-run code to include a legacy mode to emulate the behavior of the old version (v2).

Comment (plexoos) [reply StRoot/PWGTools/BadRunQA/README.md]: The requirements.txt file is platform-agnostic and used by Python package managers to establish dependencies in virtual environments. However, it seems to have been left out of this particular pull request.

**Review by plexoos — APPROVED** [review]

File: StRoot/PWGTools/BadRunQA/README.md (hunk 1 above)
Comment (ssedd1123) [reply StRoot/PWGTools/BadRunQA/README.md]: I tried to use requirements.txt on my Windows desktop and Windows laptop with pip3 install -r requirements.txt and the package "rupture" gives me errors. Something along the line of "error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools"". I did update my Microsoft C++ Build Tools but it's still not working. I can still get that particular package to work on Windows, but only if I install with conda, i.e. conda install ruptures. 

However, this requirements.txt definitely works on RCAS.

**Review by marrbnl — APPROVED** [review]

**Review by marrbnl — APPROVED** [review]

**Comment by marrbnl** [comment]
@plexoos Hello Dmitri. This pull request has been approved by two people, so I think it is ready to be merged. Since I have not done this before, should I click "Squash and merge" or something else?Thanks.

**Comment by starsdong** [comment]
Rongrong, yes. Generally, I would wait for the CI build and test jobs to finish successfully. Then the "Squash and merge" button will turn green. I guess it is not relevant for this PR anyway. We can merge now if it is urgent or wait for the rest to finish.

**Comment by marrbnl** [comment]
Thanks for the clarification, Xin. We can wait for the test to finish.

**Pull request merged**